### PR TITLE
More friendly core number requirement prompts

### PR DESCRIPTION
**Pull request by elderhammer**
This PR provides a more friendly prompt for core number requirements.
#2893

### CODE REVIEW

File: cli/src/commands/start.rs [hunk 1]
```diff
@@ -444,7 +444,10 @@ impl Start {
         // Determine the number of main cores.
         let main_cores = match num_cores {
             // Insufficient
-            0..=3 => unreachable!("The number of cores is insufficient"),
+            0..=3 => {
+                println!("The number of cores is insufficient, at least 4");
+                std::thread::exit(1);
```
Comment (ljedrz) [inline cli/src/commands/start.rs]:
```suggestion
                std::process::exit(1);
```

File: cli/src/commands/start.rs [hunk 2]
```diff
@@ -444,7 +444,10 @@ impl Start {
         // Determine the number of main cores.
         let main_cores = match num_cores {
             // Insufficient
-            0..=3 => unreachable!("The number of cores is insufficient"),
+            0..=3 => {
+                println!("The number of cores is insufficient, at least 4");
```
Comment (ljedrz) [inline cli/src/commands/start.rs]:
```suggestion
                println!("The number of cores is insufficient, at least 4 are needed.");
```

File: cli/src/commands/start.rs (hunk 1 above)
Comment (elderhammer) [reply cli/src/commands/start.rs]: Sorry, I just wrote this line of code based on my guess and didn't check docs.rs carefully. Thanks for your correction.

File: cli/src/commands/start.rs [hunk 3]
```diff
@@ -444,7 +444,10 @@ impl Start {
         // Determine the number of main cores.
         let main_cores = match num_cores {
             // Insufficient
-            0..=3 => unreachable!("The number of cores is insufficient"),
+            0..=3 => {
+                println!("The number of cores is insufficient, at least 4 are needed.");
```
Comment (howardwu) [inline cli/src/commands/start.rs]:
```suggestion
                eprintln!("The number of cores is insufficient, at least 4 are needed.");
```

**Review by howardwu — APPROVED** [review]

**Comment by elderhammer** [comment]
@all-contributors please add @elderhammer  for code,bug

**Pull request merged**